pages/posts/website-analytics-snapshot-2023.md

```diff
@@ -0,0 +1,171 @@
+
+Where are these requests for Hyper Schema meta-schemas coming from? While the physical location is uninteresting, we can look at the user agent provided. The top two, which covers 7.5 million requests, doesn't really help… `python-requests`. At most, we can see that version 2.27 and 2.28 are roughly in equal use. After that, we jump down to 635k requests, from something identifying as `neptune-client`. Curiosity piqued!
+
+They call it an "Experiment tracking tool and model registry". This is to do with Machine Learning (ML) Ops (MLOps), built for data scientists and ML engineers. I know very little about ML, so I asked my colleague to share his take on the Neptune.ai service, who knows vastly more about this space…
+
+"Piece of software that, when you tweak something about a machine learning model, let's you run some calculations that tell you how your changes affect how good the model is" - [Julian Berman](https://mastodon.social/@JulianWasTaken)
+
+Well, that's dumbed down enough for me! Thanks Julian!
+
+I guess we should reach out to them to find out about their use case for Hyper Schema!
+
+## OK, so who's still accessing `/schema`?
+
+Accessing the latest meta-schema published using this unversioned URI is not advised. We started making people aware of this about four years ago, and you'll see [a message with your redirection](https://github.com/json-schema-org/json-schema-org.github.io/blob/fb3b1e681e273738f38cb1438b0635b6d4ad5383/_layouts/redirect.html#L13) if using a browser.
+
+The vast majority of requests to this have no referring source, so that's not very helpful. Turning to user agent, we see the vast majority of requests have a user agent of ""... yup, blank. This is poor form, and I had a suspicion about what might be causing this.
+
+When you look at the adjacent data, seeing over half of the remaining top 15 user agents, it's pretty clear that most of these requests are coming from IDEs and code editors. We see InteliJ IDEA, WebStorm, and multiple versions of each. But, what we don't see is VSCode.
+
+Sure enough, VSCode, as far as we can tell, [doesn't provide a user agent](https://github.com/microsoft/vscode/issues/188398). At least, when making requests related to JSON Schema. While we love VSCode and the baked-in support for JSON Schema, there are a number of things we wish were a little better, and this is one of them.
+
+Visual Studio provides a user agent (and we'll get to that analysis later), but given Visual Studio Code does not, we can't differentiate between requests it is making, and any others which aren't providing a user agent. (We generally see a user agent is given.)
+
+## Who is accessing draft-04?
+
+The total number of requests to the draft-04 meta-schema is 7.21 million in the last 7 days.
+
+The top physical locations showed one surprise, with Peru taking the top spot, over the UK and Russia. Otherwise, the USA takes top spot with 5.73 million of those 7.21 million requests.
```

Comment:
This is a bit redundant with "top" being present three times.

```suggestion
The physical locations showed one surprise, with Peru taking the lead over the UK and Russia. Otherwise, the USA takes top spot with 5.73 million of those 7.21 million requests.
```

It's also a bit confusing.  Is the order Peru, UK, Russia, USA?  If so, is USA really taking a "top spot"?

Reply:
You're right. It would have made more senes without the comma, but I agree it's not clear.

Reply:
It still feels clunky calling out USA as the "top spot" when you've just mentioned three countries that are above it.

Maybe

> The USA follows them up with...

Reply:
That's a fair comment. I'll re-work